### PR TITLE
[v5] Remove Devtools warning

### PR DESCRIPTION
**Pull request by charkour**
For #2138

## Related Issues or Discussions

Fixes #2404
Related #2405 

## Summary

Using the devtools middleware will warn in all non-browser environments during development. This was added many years ago and likely can be removed. The docs are updated to link to the Redux DevTools Chrome extension. 

## Check List

- [x] `yarn run prettier` for formatting code and docs

### CODE REVIEW

File: tests/devtools.test.tsx [hunk 1]
```diff
@@ -154,11 +154,6 @@ describe('If there is no extension installed...', () => {
...
   })
 
-  it('[DEV-ONLY] warns if enabled in dev mode', async () => {
-    createStore(devtools(() => ({ count: 0 }), { enabled: true }))
-    expect(console.warn).toBeCalled()
-  })
-
   it.skip('[PRD-ONLY] does not warn if not in dev env', async () => {
     createStore(devtools(() => ({ count: 0 })))
     expect(console.warn).not.toBeCalled()
```
Comment (dai-shi) [inline tests/devtools.test.tsx:159]: We don't need it too.
```suggestion
```

**Review by dai-shi — COMMENTED** [review]
Thanks for working on this!

Can you remove "if not enabled" in https://github.com/pmndrs/zustand/pull/2466/files#diff-ed83d433b441e4c067aeb456c22886961dda4161280d2fbe2167d98bccf227b3R152 ?

**Comment by dai-shi** [comment]
Can you remove "if not enabled" in https://github.com/pmndrs/zustand/pull/2466/files#diff-ed83d433b441e4c067aeb456c22886961dda4161280d2fbe2167d98bccf227b3R152 ?

**Comment by charkour** [comment]
Yes I will do that now.

**Comment by charkour** [comment]
@dai-shi, this is ready for a re-review. Thanks

**Review by dbritto-dev — APPROVED** [review]

File: tests/devtools.test.tsx [hunk 2]
```diff
@@ -148,26 +148,6 @@ describe('If there is no extension installed...', () => {
       createStore(devtools(() => ({ count: 0 })))
     }).not.toThrow()
   })
```
Comment (dai-shi) [inline tests/devtools.test.tsx:150]: I mean the test itself is fine.

```suggestion
  })
  
    it('does not warn', async () => {
    createStore(devtools(() => ({ count: 0 })))
    expect(console.warn).not.toBeCalled()
  })
```

Comment (dai-shi) [reply tests/devtools.test.tsx:150]: @charkour I do not have a permission to add a commit.

Comment (charkour) [reply tests/devtools.test.tsx:150]: Sorry I misunderstood. I'll make it editable. Thank you.

Comment (charkour) [reply tests/devtools.test.tsx:150]: Oh, I found out that my personal organization to store forks means I cannot make it editable. 

I've updated the name of the test as requested. Thank you

**Review by dbritto-dev — APPROVED** [review]
LGTM

**Review by dai-shi — APPROVED** [review]
Thanks for your contribution!

**Pull request merged**